Sources/OAuthenticator/Models.swift

```diff
@@ -5,6 +5,7 @@ import Foundation
 /// This is used to abstract the actual networking system from the underlying authentication
 /// mechanism.
 public typealias URLResponseProvider = (URLRequest) async throws -> (Data, URLResponse)
+public typealias URLUserDataProvider<UserDataType: Sendable> = (URLRequest) async throws -> (UserDataType, URLResponse)
```

Comment:
You don't think "Response" was the right term?

Reply:
I've been thinking about this for a while, and also imagined a future use where uploading data to OAuth protected servers can be supported. To minimize the impact on the existing API, I would suggest having two providers. One used exclusively for authentication purposes (and could still use the name "Response"), and another for the user data transfer part. The former doesn't need to be generic, while the latter should support URLs, dates, and other types. An upload function (similar to `Authenticator.response(for:)`) would probably also benefit from a different name.

I'm fine with "Response", I was just thinking of another name for the general data type that could be used for downloading and uploading.